pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/PulsarAuthorizationProvider.java

```diff
@@ -392,23 +392,6 @@ public CompletableFuture<Boolean> checkPermission(TopicName topicName, String ro
                                 return true;
                             }
                         }
-
```

Comment:
You cannot remove this code, we need to do compatibility here. For the old version, we grant the permissions for each partitioned topic name, right now we only grant the permissions for the base partitioned topic name.

You should check whether the topic is a partitioned topic. If true, we need to check the permissions by the full name of each partition topic.

Assume the `persistent://my-tenant/my-namespace/my-topic` is partitioned topic, and has two partition:

- `persistent://my-tenant/my-namespace/my-topic-partition-0`
- `persistent://my-tenant/my-namespace/my-topic-partition-1`

You first need to get the full name of each partition topic, then check the permissions of these topic names.

Reply:
Do you mean that we should guarantee the compatibility for this situation?

- someone only grant permission for persistent://test/test/test-partition-5, but not grant permission for persistent://test/test/test
- he should still can produce msg to persistent://test/test/test-partition-5, but can not produce msg to persistent://test/test/test

Reply:
I mean that both can produce messages.

Reply:
> I mean that both can produce messages.

Actually in pulsar old version, this situation can only produce to one partition.

If we just authorized "pulsar-admin topics permissions persistent://my-tenant/my-namespace/my-topic-partition-0" in old version, only the 0 partition has permission, and the base topic has no permission, therefore msg can only be produced to partition 0, but cannot be produced to base topic.

So if we do compatible with the behavior of the old version, it should only can produce to partition 0, but not have permission to produce to base topic? Or do you think that both should be able to produced in the new version?

Besides, If we authorized "pulsar-admin topics permissions persistent://my-tenant/my-namespace/my-topic" in old version, both the base topic and the two partitions will have permissions. Then this pr use topicName.getPartitionedTopicName() to grant, revoke and check. No matter whether the request is the partition "my-topic-partition-0" or the base topic "my-topic", it will be mapped to base topic "my-topic". Since old version has base topic permission, I think this can satisfy the compatibility with old version.

Thank you for your review and Please take a look about it

Reply:
@TakaHiR07 There are five scenarios:

1. Grant permission
Based on `topicName.getPartitionedTopicName()`

2. Revoke permission
Based on `topicName.getPartitionedTopicName()`, and full name of  each partition topic 

4. Get permission
Based on `topicName.getPartitionedTopicName()`, and full name of  each partition topic 

5. Check permission
Based on `topicName.getPartitionedTopicName()`, and full name of  each partition topic 

Using the above rule can meet the old and new versions, no matter how the user grants permission.

Reply:
OK, I would update this pr soon. Thanks a lot